src/Config/Validator/Deploy/ReportDirNestingLevel.php

```diff
@@ -0,0 +1,211 @@
+            );
+        } catch (FileSystemException $exception) {
+            return $this->resultFactory->error($exception->getMessage());
+        } catch (NotEncodableValueException $exception) {
+            $message = sprintf('Config of the file %s is invalid. ', $reportConfigFile);
+            $message .= $exception->getMessage();
+            return $this->resultFactory->error(
+                $message,
+                'Fix the configuration of the directories nesting level for error reporting '
+                . 'in the file ' . $reportConfigFile
+            );
+        } catch (ValidatorException $exception) {
+            return $this->resultFactory->error($exception->getMessage(), $exception->getSuggestion());
+        }
+    }
+
+    /**
+     * Validates a value of the env variable MAGE_ERROR_REPORT_DIR_NESTING_LEVEL
+     *
+     * @param string|int $envVarReportDirNestingLevel
+     * @throws ValidatorException
+     * @return void
+     */
+    private function validateEnvVarReportDirNestingLevel($envVarReportDirNestingLevel)
+    {
+        if ($this->validateReportDirNestingLevel($envVarReportDirNestingLevel)) {
+            return;
+        }
+        throw new ValidatorException(
+            sprintf(
+                'The value `%s` of the environment variable `%s` is invalid.',
+                $envVarReportDirNestingLevel,
+                Environment::ENV_MAGE_ERROR_REPORT_DIR_NESTING_LEVEL
+            ),
+            sprintf(
+                'The value of the environment variable `%s` must be an integer between 0 and 32',
+                Environment::ENV_MAGE_ERROR_REPORT_DIR_NESTING_LEVEL
+            )
+        );
+    }
+
+    /**
+     * Validates a value of the property `config.report.dir_nesting_level`
+     * from the file <magento_root>/pub/errors/local.xml
+     *
+     * @param string|int $reportConfigDirNestingLevel
+     * @throws ValidatorException
+     * @return void
+     */
+    private function validateReportConfigDirNestingLevel($reportConfigDirNestingLevel)
+    {
+        if ($this->validateReportDirNestingLevel($reportConfigDirNestingLevel)) {
+            return;
+        }
+        throw new ValidatorException(
+            sprintf(
+                'The value `%s` of the property `config.report.dir_nesting_level` '
+                . 'defined in <magento_root>/pub/errors/local.xml is invalid',
+                $reportConfigDirNestingLevel
+            ),
+            'The value of the property `config.report.dir_nesting_level` must be an integer between 0 and 32'
+        );
+    }
+
+    /**
+     * Validates a value of the directories nesting level
+     *
+     * @param string|int $value
+     * @return bool
+     */
+    private function validateReportDirNestingLevel($value)
+    {
+        $valueInt = (int)$value;
+        return (bool)preg_match('/^([0-9]{1,2})$/', (string)$value) && (0 <= $valueInt) && (32 >= $valueInt);
```

Comment:
why do you need this preg_match?

Reply:
@NadiyaS  because in another case, values ​​like "some text", "2d", "d2" will be true.
See the test https://github.com/magento/ece-tools/pull/601/files#diff-f35961c9df30211dd545fc6a03d1b563R138

Reply:
isn't `is_numeric()` work for it?